### PR TITLE
refactor: `bucketExists`

### PR DESCRIPTION
**Pull request by trim21**

### CODE REVIEW

**Review by aldy505 — COMMENTED** [review]
LGTM

**Comment by trim21** [comment]
> Oh anyway, quick question. Why didn't this PR update the `minio.d.ts` file like other PR did? The `bucketExists` signature should be deleted, right?

I forget this

File: examples/bucket-exists.js [hunk 1]
```diff
@@ -25,11 +25,7 @@ const s3Client = new Minio.Client({
   secretKey: 'YOUR-SECRETACCESSKEY',
 })
 
-s3Client.bucketExists('my-bucketname', function (err, exists) {
```
Comment (prakashsvmx) [inline examples/bucket-exists.js:28]: Please add `.mjs` extension to the file. @trim21

**Review by prakashsvmx — APPROVED** [review]

File: examples/bucket-exists.js (hunk 1 above)
Comment (prakashsvmx) [reply examples/bucket-exists.js:28]: Also please  update README.md file for references.

**Review by aldy505 — COMMENTED** [review]
LGTM

**Review by prakashsvmx — DISMISSED** [review]

**Review by prakashsvmx — DISMISSED** [review]

File: src/minio.js [hunk 2]
```diff
@@ -1846,10 +1820,13 @@ Client.prototype.selectObjectContent = promisify(Client.prototype.selectObjectCo
 
 // refactored API use promise internally
 Client.prototype.makeBucket = callbackify(Client.prototype.makeBucket)
+Client.prototype.bucketExists = callbackify(Client.prototype.bucketExists)
 Client.prototype.removeBucket = callbackify(Client.prototype.removeBucket)
 Client.prototype.listBuckets = callbackify(Client.prototype.listBuckets)
 
 Client.prototype.statObject = callbackify(Client.prototype.statObject)
+Client.prototype.removeObject = callbackify(Client.prototype.removeObject)
```
Comment (kaankabalak) [inline src/minio.js:1828]: Looks like this is a duplicate of line 1832, is this intentional?

File: src/minio.js [hunk 3]
```diff
@@ -1817,10 +1791,13 @@ Client.prototype.selectObjectContent = promisify(Client.prototype.selectObjectCo
...
```
Comment (trim21) [inline src/minio.js:1799]:
```suggestion
```

File: src/minio.js (hunk 2 above)
Comment (trim21) [reply src/minio.js]: a mistake

**Comment by trim21** [comment]
@kaankabalak

**Review by kaankabalak — APPROVED** [review]
LGTM

**Pull request merged**